agent/acs/client/acs_client_types.go

```diff
@@ -45,18 +42,28 @@ func init() {
 		ecsacs.InactiveInstanceException{},
 	}
 
+	acsTypeMappings = make(map[string]reflect.Type)
 	// This produces a map of:
 	// "MyMessage": TypeOf(ecsacs.MyMessage)
-	typeMappings = make(map[string]reflect.Type)
 	for _, recognizedType := range recognizedTypes {
-		typeMappings[reflect.TypeOf(recognizedType).Name()] = reflect.TypeOf(recognizedType)
+		acsTypeMappings[reflect.TypeOf(recognizedType).Name()] = reflect.TypeOf(recognizedType)
 	}
 }
 
-func NewOfType(acsType string) (interface{}, bool) {
-	rtype, ok := typeMappings[acsType]
+// typeMappings implements wsclient.TypeMappings.
```

Comment:
Any reason for both 'wsclient.TypeMappings' and 'wsclient.TypeDecoder'?

Based on this, it seems like they're really tightly coupled enough that they could/should be a single interface (decoder makes sense to me). One of them is giving the types that the other will handle.